File: de.fu_berlin.inf.dpp.ui/src/de/fu_berlin/inf/dpp/ui/browser_functions/CloseSessionInvitationWizard.java
package de.fu_berlin.inf.dpp.ui.browser_functions;

import de.fu_berlin.inf.dpp.HTMLUIContextFactory;
import de.fu_berlin.inf.dpp.ui.ide_embedding.DialogManager;
import de.fu_berlin.inf.dpp.ui.pages.SessionWizardPage;

/** Close an open {@link SessionWizardPage} dialog. */
public class CloseSessionInvitationWizard extends TypedJavascriptFunction {

  public static final String JS_NAME = "closeStartSessionWizard";

  private final DialogManager dialogManager;

  /**
   * Created by PicoContainer
   *
   * @param dialogManager
   * @see HTMLUIContextFactory
   */
  public CloseSessionInvitationWizard(DialogManager dialogManager) {
    super(JS_NAME);
    this.dialogManager = dialogManager;
  }

  /** Close an open {@link SessionWizardPage} dialog. */
  @BrowserFunction
  public void closeStartSessionWizard() {
    dialogManager.closeDialogWindow(SessionWizardPage.HTML_DOC_NAME);
  }
}

File: de.fu_berlin.inf.dpp.ui/src/de/fu_berlin/inf/dpp/ui/browser_functions/CloseSessionInvitationWizard.kt
package de.fu_berlin.inf.dpp.ui.browser_functions

import de.fu_berlin.inf.dpp.HTMLUIContextFactory
import de.fu_berlin.inf.dpp.ui.ide_embedding.DialogManager
import de.fu_berlin.inf.dpp.ui.pages.SessionWizardPage

/**
 * Close an open {@link SessionWizardPage} dialog.
 *
 * Created by PicoContainer
 *
 * @param dialogManager
 * @see HTMLUIContextFactory
 */
class CloseSessionInvitationWizard (dialogManager: DialogManager?) : TypedJavaScriptFunction(JS_NAME) {
	private val dialogManager: DialogManager? = dialogManager

	/**
	 * Close an open {@link SessionWizardPage} dialog.
	 */
	@BrowserFunction
	fun closeStartSessionWizard() {
		dialogManager!!.closeDialogWindow(SessionWizardPage.HTML_DOC_NAME)
	}

	companion object {
		val JS_NAME = "closeStartSessionWizard"
	}
}